ground/src/main/java/com/google/android/ground/persistence/remote/firebase/schema/SubmissionConverter.kt

```diff
@@ -85,18 +85,23 @@ internal object SubmissionConverter {
   }
 
   private fun putValue(taskId: String, job: Job, obj: Any, data: MutableMap<String, Value>) {
-    val task = job.getTask(taskId)
-    when (task.type) {
-      Task.Type.PHOTO,
-      Task.Type.TEXT -> putTextResponse(taskId, obj, data)
-      Task.Type.MULTIPLE_CHOICE -> putMultipleChoiceResponse(taskId, task.multipleChoice, obj, data)
-      Task.Type.NUMBER -> putNumberResponse(taskId, obj, data)
-      Task.Type.DATE -> putDateResponse(taskId, obj, data)
-      Task.Type.TIME -> putTimeResponse(taskId, obj, data)
-      Task.Type.DROP_PIN -> putDropPinTaskResult(taskId, obj, data)
-      Task.Type.DRAW_AREA -> putDrawAreaTaskResult(taskId, obj, data)
-      Task.Type.CAPTURE_LOCATION -> putCaptureLocationResult(taskId, obj, data)
-      else -> throw DataStoreException("Unknown type " + task.type)
+    try {
+      val task = job.getTask(taskId)
```

Comment:
Generally, we propagate "not found" by returning `null`. Any reason not to do that here? It would also make this code a bit cleaner:

```kt
val task = job.getTask(taskId) ?: return Timber.d(e, "Invalid task ID $taskId")
```

Reply:
I agree that we can probably just return nulls here unless we really think this is an exceptional circumstance,

That said, this would be a significant behavioral change, since callers currently expect this method to return unit and throw exceptions—I'd rather keep this change separate from the fix so I've added an issue and a TODO. Thanks!

https://github.com/google/ground-android/issues/2216